it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionControllerISpec.scala

```diff
@@ -84,90 +87,249 @@ class SubscriptionControllerISpec extends BaseISpec with SessionDataMissingSpec
     }
   }
 
-  "showSubscriptionComplete" should {
-    behave like anAgentAffinityGroupOnlyEndpoint(request => controller.showSubscriptionComplete(request))
-    behave like aPageWithFeedbackLinks(
-      request => {
-        controller.showSubscriptionComplete(request)
-      },
-      authenticatedAs(subscribingCleanAgentWithoutEnrolments).withFlash("arn" -> "AARN0000001", "agencyName" -> "My Agency")
-    )
+  "showLinkAccountType (GET /link-account)" should {
+    trait RequestAndResult {
+      val request = authenticatedAs(subscribingAgentEnrolledForHMRCASAGENT)
+        .withSession("arn" -> "AARN0000001")
+      val result = await(controller.showLinkAccount(request))
+      val doc = Jsoup.parse(bodyOf(result))
+    }
+
+    behave like anAgentAffinityGroupOnlyEndpoint(controller.showLinkAccount(_))
+
+    "contain page titles and content" in new RequestAndResult {
+      result should containMessages(
+        "linkAccount.title",
+        "linkAccount.p1",
+        "linkAccount.p2",
+        "linkAccount.bullet-list.1",
+        "linkAccount.bullet-list.2",
+        "linkAccount.p3",
+        "linkAccount.p4",
+        "linkAccount.legend",
+        "linkAccount.option.yes",
+        "linkAccount.option.no"
+      )
+    }
+
+    "contain radio options for Yes and No" in new RequestAndResult {
```

Comment:
Some unit tests for form validation would be nice...

Reply:
These are the tests for the GET request so are just checking the rendered HTML has the expected content. The checks for the form validation are in the tests for the POST endpoint below